### PR TITLE
TCK Migration : Userguide & assertions 

### PR DESCRIPTION
**Pull request by alwin-joseph**
Issue https://github.com/eclipse-ee4j/jaxrs-api/issues/1040 

- Assertion files , release notes are for the 3.0 release
- Initial update to Userguide (pdf & html) sources are done. More changes can be done later on this.
- The ts.jtx exclude list file is included as a document 

To build the userguide  : 
```
cd jaxrs-tck-docs/userguide
mvn
```

### CODE REVIEW

File: jersey-tck/pom.xml [hunk 1]
```diff
@@ -304,7 +305,7 @@
                         <configuration>
                             <dependenciesToScan>jakarta.ws.rs:jakarta.ws.rs-tck</dependenciesToScan>
                             <systemPropertyVariables>
```
Comment (jansupol) [inline jersey-tck/pom.xml:307]: This change does not belong to this PR, it is about the user guide.

**Comment by jansupol** [comment]
The assertion lists are great, but we would need the ones in xml format, too. 

Should we add the new module to the pom.xml?

File: jaxrs-tck-docs/userguide/pom.xml [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+    SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+
+-->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>org.eclipse.ee4j</groupId>
+        <artifactId>project</artifactId>
+        <version>1.0.5</version>
```
Comment (jansupol) [inline jaxrs-tck-docs/userguide/pom.xml:25]: I am not sure 1.0.5 still works properly regarding the release. I'd use 1.0.7, the latest

**Comment by alwin-joseph** [comment]
> The assertion lists are great, but we would need the ones in xml format, too.
> 
I have added the xml files too. Both xml & html were updated manually for Jaxrs 3.0 release as it exists in the jakartaee-tck repository. 

> Should we add the new module to the pom.xml?

The tck userguide can be built using `mvn` from jaxrs-tck-docs/userguide folder and the same can be added as module to parent pom if necessary. That would cause tck userguide to be built everytime the maven build starts from parent pom, eventhough tck userguide module does not depend on other modules. If that is ok, I can add jaxrs-tck-docs/userguide module to pom.xml.

**Comment by spericas** [comment]
@alwin-joseph Why can't we set the version to be 3.1 as part of this migration process?

**Comment by alwin-joseph** [comment]
> @alwin-joseph Why can't we set the version to be 3.1 as part of this migration process?

I will update the Release notes with 3.1. Can the spec team help update spec/api assertions.

**Comment by jansupol** [comment]
The new assertions in the javadoc assertion list are great, but the assertion list does contain some oddities. 
For instance: JAXRS:JAVADOC:1__OLD and JAXRS:JAVADOC:1. These are basically the same, with a minor change in javadoc wording. Possibly, the XML should be updated to contain only one (the new wording) and the OLD wording is not included. The same with all other OLD assertion IDs.

The Spec Assertion list newly contains JAX-RS:SPEC assertions, whereas the old list contains JAXRS:SPEC assertions. The tests contain the old JAXRS:SPEC notion. Should we need test coverage, the test coverage tool matches the assertion ids.

**Comment by jansupol** [comment]
The documentation PDF is nicely generated, that's cool. However, the result of the maven execution is not a maven artifact that can be staged into the maven repository. Possibly, a bundle task or module would be needed. I am not sure whether as a part of this PR or a separate PR. Personally, I'd love to see it being a part of the TCK bundle rather than a separate maven artifact. What do people think?

**Comment by alwin-joseph** [comment]
> The documentation PDF is nicely generated, that's cool. However, the result of the maven execution is not a maven artifact that can be staged into the maven repository. Possibly, a bundle task or module would be needed. I am not sure whether as a part of this PR or a separate PR. Personally, I'd love to see it being a part of the TCK bundle rather than a separate maven artifact. What do people think?

1. The Ballot spec PR expects the TCK as a zip bundle of the format jakarta-restful-ws-tck-x.y.z.zip as it was done for previous releases. 
2. What we can do is to generate the zip TCK bundle that contains the TCK jar jakarta.ws.rs-tck-x.y.z.jar and a documentation folder (that contains userguide, assertions etc ). This can be done as a part of the CI job https://ci.eclipse.org/jakartaee-tck/job/jakarta-rest-tck-build/ which currently adds only the TCK jar within the zip (https://ci.eclipse.org/jakartaee-tck/job/jakarta-rest-tck-build/lastSuccessfulBuild/artifact/jaxrs-tck/target/jakarta.ws.rs-tck-3.1.0.zip).
 
Would that suffice?

**Comment by spericas** [comment]
> > The documentation PDF is nicely generated, that's cool. However, the result of the maven execution is not a maven artifact that can be staged into the maven repository. Possibly, a bundle task or module would be needed. I am not sure whether as a part of this PR or a separate PR. Personally, I'd love to see it being a part of the TCK bundle rather than a separate maven artifact. What do people think?
> 
> 1. The Ballot spec PR expects the TCK as a zip bundle of the format jakarta-restful-ws-tck-x.y.z.zip as it was done for previous releases.
> 2. What we can do is to generate the zip TCK bundle that contains the TCK jar jakarta.ws.rs-tck-x.y.z.jar and a documentation folder (that contains userguide, assertions etc ). This can be done as a part of the CI job https://ci.eclipse.org/jakartaee-tck/job/jakarta-rest-tck-build/ which currently adds only the TCK jar within the zip (https://ci.eclipse.org/jakartaee-tck/job/jakarta-rest-tck-build/lastSuccessfulBuild/artifact/jaxrs-tck/target/jakarta.ws.rs-tck-3.1.0.zip).
> 
> Would that suffice?

AFAIK, yes, that would be sufficient.

**Comment by edbratt** [comment]
FWIW, I've posed a question to the Spec. Committee, asking if we can relax the submission requirements and allow EITHER .zip or .jar file names for the final TCK archive. Continue working toward creating a .zip for now, but if the committee will allow it, I think it will be better in the long run to allow .jar since it can be used more readily by the intended TCK users.

**Comment by alwin-joseph** [comment]
> The new assertions in the javadoc assertion list are great, but the assertion list does contain some oddities. For instance: JAXRS:JAVADOC:1__OLD and JAXRS:JAVADOC:1. These are basically the same, with a minor change in javadoc wording. Possibly, the XML should be updated to contain only one (the new wording) and the OLD wording is not included. The same with all other OLD assertion IDs.
> 
Removed the *__OLD javadocs from both xml & html javadoc assertion files in 3.1.0

> The Spec Assertion list newly contains JAX-RS:SPEC assertions, whereas the old list contains JAXRS:SPEC assertions. The tests contain the old JAXRS:SPEC notion. Should we need test coverage, the test coverage tool matches the assertion ids.
JAX-RS:SPEC is renamed to JAXRS:SPEC for spec assertion files in 3.1.0

File: jaxrs-tck-docs/assertions/JAXRSJavadocAssertions_3.0.0.html [hunk 3]
```diff
@@ -0,0 +1,9248 @@
+<html xmlns:Assertions="https://raw.githubusercontent.com/eclipse-ee4j/jakartaee-tck/master/internal/docs">
+<head>
+<META http-equiv="Content-Type" content="text/html; charset=UTF-8">
+<!--
+
+    Copyright (c) 2018, 2020 Oracle and/or its affiliates. All rights reserved.
```
Comment (gurunrao) [inline jaxrs-tck-docs/assertions/JAXRSJavadocAssertions_3.0.0.html:6]: we can change year to just 2021, since files are new to repository.

File: jaxrs-tck-docs/ts.jtx [hunk 4]
```diff
@@ -0,0 +1,82 @@
+#
```
[the remark this answers is not in the document]
Comment (alwin-joseph) [reply jaxrs-tck-docs/ts.jtx:1]: Sure. Any suggestions for a new exclude list filename ? This will have to be changed in docs as well.

**Review by spericas — APPROVED** [review]

**Review by jansupol — APPROVED** [review]

**Review by mkarg — COMMENTED** [review]
I am generally +1 for this request, but abstaining from an official vote, as I am not deeply familiar with the Arquillian and the TCK yet. So I kindly request votes from committers with more experience in that area.
